testing/src/main/java/io/stargate/it/http/RestApiv2Test.java

```diff
@@ -2350,6 +2350,302 @@ public void deleteColumnPartitionKey() throws IOException {
+
+    RestUtils.post(
+        authToken,
+        String.format("%s:8082/v2/schemas/keyspaces/%s/types", host, keyspaceName),
+        udtString,
+        HttpStatus.SC_CREATED);
+
+    // throws error because same name, but ifNotExists = false
+    String response =
+        RestUtils.post(
+            authToken,
+            String.format("%s:8082/v2/schemas/keyspaces/%s/types", host, keyspaceName),
+            udtString,
+            HttpStatus.SC_BAD_REQUEST);
+
+    String typeName = "udt1";
+
+    Error error = objectMapper.readValue(response, Error.class);
+    assertThat(error.getCode()).isEqualTo(HttpStatus.SC_BAD_REQUEST);
+    assertThat(error.getDescription())
+        .isEqualTo(
+            String.format(
+                "Bad request: A type named \"%s\".%s already exists", keyspaceName, typeName));
+
+    // don't create and don't throw exception because ifNotExists = true
+    udtString =
+        "{\"name\": \"udt1\", \"ifNotExists\": true, \"fieldDefinitions\":[{\"name\":\"firstname\",\"typeDefinition\":\"text\"}]}";
+    RestUtils.post(
+        authToken,
+        String.format("%s:8082/v2/schemas/keyspaces/%s/types", host, keyspaceName),
+        udtString,
+        HttpStatus.SC_CREATED);
+  }
+
+  @Test
+  public void updateInvalidUdt() throws IOException {
+    createKeyspace(keyspaceName);
+
+    String udtString =
+        "{\"name\": \"udt1\", \"fieldDefinitions\":[{\"name\":\"firstname\",\"typeDefinition\":\"text\"}]}";
+
+    RestUtils.post(
+        authToken,
+        String.format("%s:8082/v2/schemas/keyspaces/%s/types", host, keyspaceName),
+        udtString,
+        HttpStatus.SC_CREATED);
+
+    // add existing field
+    udtString =
+        "{\"name\": \"udt1\", \"add-type\":[{\"name\":\"firstname\",\"typeDefinition\":\"text\"}]}";
+
+    RestUtils.put(
+        authToken,
+        String.format("%s:8082/v2/schemas/keyspaces/%s/types", host, keyspaceName),
+        udtString,
+        HttpStatus.SC_BAD_REQUEST);
+
+    // missing add-type and rename-type
+    udtString =
+        "{\"name\": \"udt1\", \"fieldDefinitions\":[{\"name\":\"firstname\",\"typeDefinition\":\"text\"}]}";
+
+    RestUtils.put(
+        authToken,
+        String.format("%s:8082/v2/schemas/keyspaces/%s/types", host, keyspaceName),
+        udtString,
+        HttpStatus.SC_BAD_REQUEST);
+  }
+
+  @Test
+  public void updateUdt() throws IOException {
+    createKeyspace(keyspaceName);
+
+    String udtString =
+        "{\"name\": \"udt1\", \"fieldDefinitions\":[{\"name\":\"firstname\",\"typeDefinition\":\"varchar\"}]}";
+
+    RestUtils.post(
+        authToken,
+        String.format("%s:8082/v2/schemas/keyspaces/%s/types", host, keyspaceName),
+        udtString,
+        HttpStatus.SC_CREATED);
+
+    udtString =
+        "{\"name\": \"udt1\", \"add-type\":[{\"name\":\"lastname\",\"typeDefinition\":\"varchar\"}]}";
```

Comment:
Again, apologies for late comments, but as for me `fieldDefinitions` looks a bit out-of-style with respect to `add-type`... Could we use hyphenation or camel-case in _both_ cases?

From my personal POV (totally optional). I'd use:
* for create: `{"name", "SomeName", "fields": {...}`
* for adding fields: `{"name", "SomeName", "add-fields": {...}`
* for renames (line 2450): `{"name", "SomeName", "rename-fields": {...}`

WDYT?